Clang analyzer: warning: comparing floating point with == or != is unsafe

I found an issue when running under clang-analyzer static code analysis. I have no real fix for this but comparision of floats is not a very good idea.

```
[ 97%] Building C object tests/CMakeFiles/test-server.dir/Unity/src/unity.c.o
cd influxdb-c/build/Debug/tests && /usr/share/clang/scan-build-3.5/ccc-analyzer  -DDEBUG -DHAVE_FCNTL_H -DHAVE_SNPRINTF -DHAVE_STDARG_H -DHAVE_STRDUP -DHAVE_STRINGS_H -DHAVE_STRNCASECMP -DHAVE_SYS_STAT_H -DHAVE_SYS_TYPES_H -DHAVE_UNISTD_H -DHAVE_VASPRINTF -DHAVE_VSNPRINTF -DSTDC_HEADERS -Iinfluxdb-c/build/Debug/src -Iinfluxdb-c/3thparty -isystem influxdb-c/3thparty/json-c -Iinfluxdb-c/build/Debug/3thparty/json-c -Iinfluxdb-c/tests/Unity/src  -std=gnu99 -Wall -Wextra -Wshadow -D_FORTIFY_SOURCE=2 -fstrict-aliasing -Wformat -Wformat-security -Wmaybe-uninitialized -Wimplicit-function-declaration -Wredundant-decls -Wundef -Wswitch -Wfloat-equal -O0 -ggdb -ftest-coverage -g   -o CMakeFiles/test-server.dir/Unity/src/unity.c.o   -c influxdb-c/tests/Unity/src/unity.c
influxdb-c/tests/Unity/src/unity.c: In function ‘UnityAssertEqualFloatArray’:
influxdb-c/tests/Unity/src/unity.c:679:26: warning: comparing floating point with == or != is unsafe [-Wfloat-equal]
         if ((diff * 0.0f != 0.0f) || (diff > tol))
                          ^
influxdb-c/tests/Unity/src/unity.c: In function ‘UnityAssertFloatsWithin’:
influxdb-c/tests/Unity/src/unity.c:722:22: warning: comparing floating point with == or != is unsafe [-Wfloat-equal]
     if ((diff * 0.0f != 0.0f) || (pos_delta < diff))
                      ^
```

--- a/src/unity.c
+++ b/src/unity.c
@@ -674,7 +674,7 @@ void UnityAssertEqualFloatArray(UNITY_PTR_ATTRIBUTE const _UF* expected,
             tol = 0.0f - tol;
 
         //This first part of this condition will catch any NaN or Infinite values
-        if (isnan(diff) || (diff > tol))
+        if (isnan(diff) || isinf(diff) || (diff > tol))
         {
             UnityTestResultsFailBegin(lineNumber);
             UnityPrint(UnityStrElement);
@@ -717,7 +717,7 @@ void UnityAssertFloatsWithin(const _UF delta,
     }
 
     //This first part of this condition will catch any NaN or Infinite values
-    if (isnan(diff) || (pos_delta < diff))
+    if (isnan(diff) || isinf(diff) || (pos_delta < diff))
     {
         UnityTestResultsFailBegin(lineNumber);
 #ifdef UNITY_FLOAT_VERBOSE
@@ -838,7 +838,7 @@ void UnityAssertEqualDoubleArray(UNITY_PTR_ATTRIBUTE const _UD* expected,
             tol = 0.0 - tol;
 
         //This first part of this condition will catch any NaN or Infinite values
-        if (isnan(diff) || (diff > tol))
+        if (isnan(diff) || isinf(diff) || (diff > tol))
         {
             UnityTestResultsFailBegin(lineNumber);
             UnityPrint(UnityStrElement);
@@ -881,7 +881,7 @@ void UnityAssertDoublesWithin(const _UD delta,
     }
 
     //This first part of this condition will catch any NaN or Infinite values
-    if (isnan(diff) || (pos_delta < diff))
+    if (isnan(diff) || isinf(diff) || (pos_delta < diff))
     {
         UnityTestResultsFailBegin(lineNumber);
 #ifdef UNITY_DOUBLE_VERBOSE

--- a/src/unity.c
+++ b/src/unity.c
@@ -668,13 +668,13 @@ void UnityAssertEqualFloatArray(UNITY_PTR_ATTRIBUTE const _UF* expected,
     {
         diff = *ptr_expected - *ptr_actual;
         if (diff < 0.0f)
-          diff = 0.0f - diff;
+            diff = 0.0f - diff;
         tol = UNITY_FLOAT_PRECISION * *ptr_expected;
         if (tol < 0.0f)
             tol = 0.0f - tol;
 
         //This first part of this condition will catch any NaN or Infinite values
-        if ((diff * 0.0f != 0.0f) || (diff > tol))
+        if (isnan(diff) || (diff > tol))
         {
             UnityTestResultsFailBegin(lineNumber);
             UnityPrint(UnityStrElement);
@@ -717,7 +717,7 @@ void UnityAssertFloatsWithin(const _UF delta,
     }
 
     //This first part of this condition will catch any NaN or Infinite values
-    if ((diff * 0.0f != 0.0f) || (pos_delta < diff))
+    if (isnan(diff) || (pos_delta < diff))
     {
         UnityTestResultsFailBegin(lineNumber);
 #ifdef UNITY_FLOAT_VERBOSE
@@ -838,7 +838,7 @@ void UnityAssertEqualDoubleArray(UNITY_PTR_ATTRIBUTE const _UD* expected,
             tol = 0.0 - tol;
 
         //This first part of this condition will catch any NaN or Infinite values
-        if ((diff * 0.0 != 0.0) || (diff > tol))
+        if (isnan(diff) || (diff > tol))
         {
             UnityTestResultsFailBegin(lineNumber);
             UnityPrint(UnityStrElement);
@@ -881,7 +881,7 @@ void UnityAssertDoublesWithin(const _UD delta,
     }
 
     //This first part of this condition will catch any NaN or Infinite values
-    if ((diff * 0.0 != 0.0) || (pos_delta < diff))
+    if (isnan(diff) || (pos_delta < diff))
     {
         UnityTestResultsFailBegin(lineNumber);
 #ifdef UNITY_DOUBLE_VERBOSE

